salesagility/SuiteCRM
# Feature Composer Extender

## Description




- Removed manually installed ```vakata/jstree``` from ```include/javascript```.
- Added composer installers-extender to allow setting custom installer paths.
- Implemented ```post-install-cmd``` and ```post-update-cmd``` composer scripts.
- Implemented  ```ComposerEvent::cleanPackages``` to remove non-dist dirs and files from installed components (tests, demos, dependency management files, etc).
- Upgraded ```vakata/jstree``` to v3, now managed by composer.
- Set composer ```preferred-install``` to dist.

## Motivation and Context

- This lets us manage front-end component packages like jstree directly in composer while also giving us the ability to clean up leftover files.

## How To Test This

1. Remove ```include/javascript/jstree```.
2. Run ```composer install``` and see that the component lib is pulled down correctly.
3. Test JStree functionality (Email folders, reports, etc).

## Types of changes


### Final checklist
